cdap-docs/developers-manual/source/building-blocks/transactional-messaging-system.rst

```diff
@@ -0,0 +1,295 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2017 Cask Data, Inc.
+    :description: Transactional Messaging System, a ACID-guaranteed "publish-and-subscribe" messaging service
+
+.. _transactional-messaging-system:
+
+==============================
+Transactional Messaging System
+==============================
+
+.. topic::  **Note: Beta Feature** 
+
+    TMS is currently a **beta** feature of CDAP |release|, and is subject to change without notice.
+
+
+Overview
+========
+The Transactional Messaging System (TMS) is a CDAP service that provides a
+"publish-and-subscribe" messaging system that understands transactions.
+
+It uses HBase for the persistent and durable storage of messages, and supports both
+transactional and non-transactional message publishing and consumption.
+
+TMS provides ACID *(atomicity, consistency, isolation,* and *durability)* guarantees,
+using `Apache Tephra <http://tephra.incubator.apache.org>`__ for the ACI portion. TMS
+doesn't interact directly with Tephra; instead, the client provides the transaction, as
+described below.
+
+TMS is similar to `Apache Kafka <https://kafka.apache.org>`__ but with added guarantees as
+to ordering and persistence of messages.
+
+
+Topics
+======
+Topics are created in TMS using both a namespace and a topic name. Note that while it is
+possible to fetch from ``system`` namespace, attempting to publish to the ``system``
```

Comment:
@chtyim Looks like we allow creation, deletion of topics in system namespace through MessagingAdmin. Should we disallow that?